src/actions/eas-update.ts

```diff
@@ -0,0 +1,88 @@
+import { getBooleanInput, getInput, info, setOutput } from '@actions/core';
+
+import { createEasQr, lastUpdate, projectDeepLink, projectInfo, projectLink, projectOwner, projectQR } from '../expo';
+import { createIssueComment, pullContext } from '../github';
+import { template } from '../utils';
+import { executeAction } from '../worker';
+
+export type UpdateInput = ReturnType<typeof updateInput>;
+
+export const DEFAULT_ID = `app:@{projectOwner}/{projectSlug} channel:{channel}`;
+
+export const DEFAULT_MESSAGE_FOR_UPDATE =
+  `This pull request was automatically deployed using [Expo GitHub Actions](https://github.com/expo/expo-github-action/tree/main/preview-comment)!\n` +
+  `\n- Project: **@{projectOwner}/{projectSlug}**` +
+  `\n- Branch: **{channel}**`;
+
+export const DEFAULT_SYSTEM_QR = `\n\n For {system}: \n <a href="{qr}"><img src="{qr}" height="200px" width="200px"></a>`;
+
+export const updateInput = () => {
+  return {
+    channel: getInput('channel') || 'default',
+    comment: !getInput('comment') || getBooleanInput('comment'),
+    message: getInput('message') || DEFAULT_MESSAGE_FOR_UPDATE,
+    messageId: getInput('message-id') || DEFAULT_ID,
+    project: getInput('project'),
+    githubToken: getInput('github-token'),
+    ios: getBooleanInput('is-ios-build') || true,
+    android: getBooleanInput('is-android-build') || true,
+  };
+};
+
+export async function updateAction(input: UpdateInput = updateInput()) {
+  const project = await projectInfo(input.project);
+  if (!project.owner) {
+    project.owner = await projectOwner();
+  }
+  if (!input.channel) {
```

Comment:
This check should be made initially, to make the action fail fast

Reply:
This one has a default value of `default` it's set on line `21`